Sources/Observation.swift

```diff
@@ -9,6 +9,21 @@ public protocol DefaultsObservation: AnyObject {
 	func removeLifetimeTie()
 }
 
+private struct DefaultsObservationAssociationKeys {
+	static let lifetimeAssociation = AssociatedObject<LifetimeAssociation>()
+}
```

Comment:
I was hoping abstracting the lifetime handling to `LifetimeAssociation` would let us not have to use associated object here.